### PR TITLE
Soft Entities

### PR DESCRIPTION
**Pull request by druiz17**
This PR enables soft-entities  by adding  the ```relayParentJoints``` for `ModelEntityItem`.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5908/app_12078_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5908/HighFidelity-Beta-PR12078-6fa3446c57d15c336c03e4982dbf2be6c00559f7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5908/HighFidelity-Beta-PR12078-6fa3446c57d15c336c03e4982dbf2be6c00559f7.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5923/app_12078_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5923/HighFidelity-Beta-PR12078-674e8e1ac277d91aa121f7d9bd07b451b4254525.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5923/HighFidelity-Beta-PR12078-674e8e1ac277d91aa121f7d9bd07b451b4254525.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5939/app_12078_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5939/HighFidelity-Beta-PR12078-527d00871a8a6d8d7bd5e55095c2e0131f1b6a88.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5939/HighFidelity-Beta-PR12078-527d00871a8a6d8d7bd5e55095c2e0131f1b6a88.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5952/app_12078_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5954/app_12078_PR.apk

**Comment by druiz17** [comment]
Test plan (2 testers minimum)
*  tester 1 - download unzip and run the script bellow - the script will add a avatar model.
* All testers should see the newly added avatar model copy the movements of tester 1. 
[soft-entity.zip](https://github.com/highfidelity/hifi/files/1604865/soft-entity.zip)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5954/HighFidelity-Beta-PR12078-597cb6ae9ed6f2d0b62d92348064c3573f7e821b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5954/HighFidelity-Beta-PR12078-597cb6ae9ed6f2d0b62d92348064c3573f7e821b.exe

File: libraries/avatars-renderer/src/avatars-renderer/Avatar.cpp [hunk 1]
```diff
@@ -347,6 +349,38 @@ void Avatar::updateAvatarEntities() {
...
 }
 
+void Avatar::relayJointDataToChildren() {
+    forEachChild([&](SpatiallyNestablePointer child) {
+        if (child->getNestableType() == NestableType::Entity) {
+            auto modelEntity = std::dynamic_pointer_cast<RenderableModelEntityItem>(child);
+            if (modelEntity) {
+                if (modelEntity->getRelayParentJoints()) {
+                    QStringList modelJointNames = modelEntity->getJointNames();
+                    QStringList avatarJointNames = getJointNames();
```
Comment (sethalves) [inline libraries/avatars-renderer/src/avatars-renderer/Avatar.cpp]: Is this too much work to be doing every frame?  Can we come up with a way to cache the joint mappings so they don't have to be recreated for each simulation loop?

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6172/app_12078_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6172/HighFidelity-Beta-PR12078-c4253f3b3b7e250ad2717353446bf39863752597.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6172/HighFidelity-Beta-PR12078-c4253f3b3b7e250ad2717353446bf39863752597.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6173/app_12078_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6173/HighFidelity-Beta-PR12078-d0bff2947158b80a597f8f76553389244df4d919.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6173/HighFidelity-Beta-PR12078-d0bff2947158b80a597f8f76553389244df4d919.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6239/app_12078_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6239/HighFidelity-Beta-PR12078-96a97b5938694c3c5dd256f03e3c8e167cd8d31f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6239/HighFidelity-Beta-PR12078-96a97b5938694c3c5dd256f03e3c8e167cd8d31f.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6263/app_12078_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6263/HighFidelity-Beta-PR12078-088f227df27181d80bd8d1d18ee03855b95fc951.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6263/HighFidelity-Beta-PR12078-088f227df27181d80bd8d1d18ee03855b95fc951.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6269/app_12078_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6269/HighFidelity-Beta-PR12078-283b9a8003458af3ef7f00f82760fcdbfb6c7caf.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6269/HighFidelity-Beta-PR12078-283b9a8003458af3ef7f00f82760fcdbfb6c7caf.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6292/app_12078_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6292/HighFidelity-Beta-PR12078-645f608814746ce1988b9e4b38abd43c68690306.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6292/HighFidelity-Beta-PR12078-645f608814746ce1988b9e4b38abd43c68690306.exe

**Comment by tapalisa** [comment]
Test plan:
- put on the Priscilla avatar
-Run Shirt.js
-check that it looks correct
-move your avatar around, make sure it continues to look correct 
-Have an observer look at you as you move and make sure it looks correct
-Remove the soft entity using the Create menu
-Make sure your avatar has gone back to normal

**Comment by QAadahlin** [comment]
Currently Testing

**Comment by QAadahlin** [comment]
@conklin94122 Completed Testing for PR#12078. Tested on both Mac and Windows. FAILED for both. The shirt.js did not render correctly on either the Mac or Windows builds. In addition, the avatar was stuck in a T-position on the Windows build after running the shirt script. Refer to images provided. 
Windows build

**Comment by QAadahlin** [comment]
@conklin94122 Completed Testing for PR#12078. Tested on both Mac and Windows. FAILED for both. The shirt.js did not render correctly on either the Mac or Windows builds. In addition, the avatar was stuck in a T-position on the Windows build after running the shirt script. Refer to images provided.
Windows build
t pose

**Comment by druiz17** [comment]
@QAadahlin the mac picture is the desired result. Can you retest the windows build. I saw no issue when I ran the latest build. If the issue happens again can you give me more description of what you did.

**Comment by QAadahlin** [comment]
Currently Re-testing

**Comment by QAadahlin** [comment]
@conklin94122 @druiz17 Completed Re-test. FAILED on Windows. The steps I took are as follows:

**Comment by QAadahlin** [comment]
@conklin94122 @druiz17 Completed Re-test. FAILED on Windows. The steps I took are as follows (after downloading the shirt.js)
1) Loaded the interface
2) Ensure the avatar used was the Photo-Real Female Avatar (Priscilla). 
3) From the interface toolbar, go to: Edit > Running Scripts
4) In the Running Scripts window, selected the Load Scripts "From Disk" button
5) Located and opened the shirt.js in the downloads folder
6) The script loaded and rendered as seen in the image for the Windows build as follows:
priscilla

**Comment by QAadahlin** [comment]
@conklin94122 @druiz17 Attached are the log files.
[hifi-log.txt](https://github.com/highfidelity/hifi/files/1637092/hifi-log.txt)

**Comment by druiz17** [comment]
@QAadahlin I think your using the wrong windows build because I see debug statements in the logs which should not be there. Please use the latest build and do a clean install https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6292/HighFidelity-Beta-PR12078-645f608814746ce1988b9e4b38abd43c68690306.exe

**Comment by QAadahlin** [comment]
Currently testing.

**Comment by QAadahlin** [comment]
@conklin94122 @druiz17 Completed Testing for PR#12078. PASSED.

**Pull request merged**